internal/controller/gcpinstall/run.go

```diff
@@ -52,23 +70,85 @@ https://cloud.google.com/iam/docs/creating-managing-service-accounts`,
 	--display-name="${SERVICE_ACCOUNT}" \
 	--description="Runs Flightcrew's Control Tower VM."`,
 		}),
-		checkIAMRole,
-		command.NewWriteModel(command.Opts{
-			SkipIfSucceed: checkIAMRole,
-			Description:   "This command creates an IAM role from `${IAM_FILE}` for the Flightcrew VM to access configs and monitoring data.\n\nhttps://cloud.google.com/iam/docs/understanding-custom-roles",
-			Command: `gcloud iam roles create ${IAM_ROLE} \${PROJECT_OR_ORG_FLAG}
-	--file=${IAM_FILE}`,
-		}),
-		command.NewWriteModel(command.Opts{
-			SkipIfSucceed: checkIAMRole,
-			Description: `This command attaches the IAM role to Flightcrew's service account, which will give the IAM permissions to a new VM.
+	}
+}
+
+func getIAMRoleCommands(args map[string]string) []*command.Model {
+	newCheckIAMRole := func(replacer *strings.Replacer) *command.Model {
+		cmd := command.NewReadModel(command.Opts{
+			Description: "Check if a ${PERMISSIONS} Flightcrew IAM Role already exists or needs to be created.",
+			Command: `gcloud iam roles describe \${PROJECT_OR_ORG_FLAG}
+	"${ROLE}" >/dev/null 2>&1`,
+			Message: map[command.State]string{
+				command.PassState: "This Flightcrew IAM role already exists.",
+				command.FailState: "No IAM role found. Next step is to create one.",
+			},
+		})
+		cmd.Replace(replacer)
+		return cmd
+	}
+	newCreateIAMRole := func(replacer *strings.Replacer, skipIfSucceed *command.Model) *command.Model {
+		cmd := command.NewWriteModel(command.Opts{
+			SkipIfSucceed: skipIfSucceed,
+			Description:   "This command creates a ${PERMISSIONS} IAM role from `${FILE}` for the Flightcrew VM to access configs and monitoring data.\n\nhttps://cloud.google.com/iam/docs/understanding-custom-roles",
+			Command: `gcloud iam roles create ${ROLE} \${PROJECT_OR_ORG_FLAG}
+	--file=${FILE}`,
+		})
+		cmd.Replace(replacer)
+		return cmd
+	}
+	newAttachPolicy := func(replacer *strings.Replacer, skipIfSucceed *command.Model) *command.Model {
```

Comment:
Do you think it feels more logical to create the check/create the IAM roles, then check/create the service account, then do the attach? It's all the same functionally, but just wondering if it would feel more right to have

1. IAM stuff - create permissions
2. Service account stuff - create account and attach IAM permissions to it
3. VM stuff - create VM and run as service account

Reply:
Agreed -- reordered, and also added additional skip commands for the bindings.